vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/BaseMO.java

```diff
@@ -153,4 +168,66 @@ public int getCustomFieldKey(String morType, String fieldName) throws Exception
 
         return cfmMo.getCustomFieldKey(morType, fieldName);
     }
+
+    private static UnmanagedInstanceTO.PowerState convertPowerState(VirtualMachinePowerState powerState) {
+        return powerState == VirtualMachinePowerState.POWERED_ON ? UnmanagedInstanceTO.PowerState.PowerOn :
+                powerState == VirtualMachinePowerState.POWERED_OFF ? UnmanagedInstanceTO.PowerState.PowerOff : UnmanagedInstanceTO.PowerState.PowerUnknown;
+    }
+
+    protected List<UnmanagedInstanceTO> convertVmsObjectContentsToUnmanagedInstances(List<ObjectContent> ocs, String keyword) throws Exception {
```

Comment:
complexity of this method is a bit high, would be nice to modularise it a bit.

Reply:
Thanks @DaanHoogland, fixed